core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/MergeLimitWithTopN.java

```diff
@@ -55,6 +56,7 @@
     private static final Capture<TopNNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
This should be allowed as long as the ordering scheme is compatible (which should always be the case unless there's a bug)

Reply:
I have allowed this now, but I've added a check to verify that source local properties are satisfying limit input ordering.
Let me know if this is fine or the verification check is unnecessary.